floss/language/go/extract.py

```diff
@@ -0,0 +1,511 @@
+# Copyright (C) 2023 Mandiant, Inc. All Rights Reserved.
+
+import re
+import sys
+import struct
+import logging
+import pathlib
+import argparse
+from typing import List, Iterable, Optional
+from itertools import chain
+
+import pefile
+
+from floss.results import StaticString, StringEncoding
+
+logger = logging.getLogger(__name__)
+
+MIN_STR_LEN = 6
+
+
+def extract_strings_from_import_data(pe: pefile.PE) -> List[StaticString]:
+    if not hasattr(pe, "DIRECTORY_ENTRY_IMPORT"):
+        # This is a Go binary, but it doesn't have an import table
+        return []
+
+    import_strings = list()
+
+    for entry in pe.DIRECTORY_ENTRY_IMPORT:
+        for imp in entry.imports:
+            if imp.name is not None:
+                addr = imp.name_offset
+                try:
+                    import_string = StaticString(
+                        string=imp.name.decode("ascii"), offset=addr, encoding=StringEncoding.ASCII
+                    )
+                    import_strings.append(import_string)
+                except UnicodeDecodeError:
+                    continue
+
+    return import_strings
+
+
+def extract_build_id(section_data, min_length) -> List[StaticString]:
+    # Build ID is a string that starts with "\xff\x20 Go build ID: " and ends with "\n"
+    # FF 20 47 6F 20 62 75 69  6C 64 20 49 44 3A 20 22  . Go build ID: "
+    # 36 4E 31 4D 77 6E 30 31  72 46 6E 41 51 4B 62 5A  6N1Mwn01rFnAQKbZ
+    # 73 46 5A 32 2F 38 41 6B  75 63 4B 46 58 4D 49 54  sFZ2/8AkucKFXMIT
+    # 63 51 52 49 75 55 5F 79  32 2F 62 76 4F 67 56 37  cQRIuU_y2/bvOgV7
+    # 52 4D 54 72 77 73 7A 5A  39 57 7A 69 6C 64 2F 72  RMTrwszZ9Wzild/r
+    # 33 31 50 47 70 61 6B 2D  48 77 36 4B 72 77 59 6E  31PGpak-Hw6KrwYn
+    # 52 4E 73 22 0A 20 FF CC  CC CC CC CC CC CC CC CC  RNs". ..........
+
+    build_id_regex = re.compile(b"(?<=\xff\x20)(?P<buildid>(.)*)\x0A")
```

Comment:
i *think* the `.*` is greedy and will consume newlines until the last newline in the section. therefore i think `\x0A` should be excluded from the pattern:


```suggestion
    build_id_regex = re.compile(b"(?<=\xff\x20)(?P<buildid>[^\x0A]*)\x0A")
```

Reply:
maybe further restrict the pattern so its less likely to match. using only two bytes to specify the pattern is more likely to false positive against random data.

```suggestion
    build_id_regex = re.compile(b"(?<=\xff\x20)(?P<buildid>Go build ID: [^\x0A]*)\x0A")
```